docs/components/tree-list.md

```diff
@@ -0,0 +1,330 @@
+# Tree List
+
+**KTreeList** - A drag-n-drop reorderable list component.
+
+::: tip
+It can be tricky to determine if an item will be places below an item as a sibling or as a child when dropping. Items to be dropped as children should be dropped near the bottom border line of the parent. If this will be the first child of an item, you will notice the margin between the two items disappearing if you are hovering in the right place.
+:::
+
+<KTreeList :items="defaultItems" />
+
+```html
+<KTreeList :items="items" />
+```
+
+## Props
+
+### items
+
+An array of items that make up the tree.
+
+<KTreeList :items="defaultItems2" />
+
+```html
+<template>
+  <KTreeList :items="items" />
+</template>
+
+<script lang="ts">
+import { defineComponent } from 'vue'
+
+export default defineComponent({
+  data() {
+    return {
+      items: [{
+        name: "Cats",
+        id: 'cats'
+      },
+      {
+        name: "Dogs",
+        id: 'dogs',
+        children: [{
+          name: "Puppies",
+          id: 'puppies'
+        }]
+      },
+      {
+        name: "Bunnies",
+        id: 'bunnies'
+      }]
+    }
+  }
+})
+</script>
+```
+
+### disableDrag
+
+Boolean (defaults to `false`) to turn of drag-n-drop reordering of the list.
+
+<KTreeList disable-drag :items="disableItems" />
+
+```html
+<KTreeList disable-drag :items="items" />
+```
+
+### maxLevel
+
+Use this prop to customize the maximum supported depth of the tree. We default to a max level of `2` or support for parents, children, and grandchildren.
+
+<KTreeList :items="maxLevelItems" :max-level="1" />
+
+```html
+<KTreeList :items="items" :max-level="1" />
+```
+
+### v-model
+
+`KTreeList` works with v-model for data binding.
+
+::: danger
+You cannot use `v-model` with the `items` prop. You must use one or the other.
+:::
+
+<div>
+  <KLabel>Value:</KLabel> {{ myList }}
+  <KTreeList v-model="myList" />
+  <br>
+  <KButton @click="reset">Reset</KButton>
+</div>
+
+```html
+<template>
+  <KLabel>Value:</KLabel> {{ myList }}
+  <KTreeList v-model="myList" />
+  <KButton @click="reset">Reset</KButton>
+</template>
+
+<script lang="ts">
+import { defineComponent } from 'vue'
+
+export default defineComponent({
+  data() {
+    return {
+      myList: [{
+        name: "Cats",
+        id: 'cats'
+      },
+      {
+        name: "Dogs",
+        id: 'dogs',
+        children: [{
+          name: "Puppies",
+          id: 'puppies'
+        }]
+      },
+      {
+        name: "Bunnies",
+        id: 'bunnies'
+      }]
+    }
+  },
+  methods: {
+    reset() {
+      this.myList = [{
+        name: "Cats",
+        id: 'cats'
+      },
+      {
+        name: "Dogs",
+        id: 'dogs',
+        children: [{
+          name: "Puppies",
+          id: 'puppies'
+        }]
+      },
+      {
+        name: "Bunnies",
+        id: 'bunnies'
+      }]
+    }
+  }
+})
+</script>
+```
+
+## KTreeItem
+
+### Properties
+
+- `name` (required) - text displayed as the label for the item
```

Comment:
I think these properties would go in the `items` prop documentation.

The `KTreeItem` component shouldn't be documented since its internal and cannot be used by a developer individually.